### PR TITLE
Bug 1369067 - file security bugs: provide option and auto-enable it based on search terms

### PR DESCRIPTION
**Pull request by Archaeopteryx**

### CODE REVIEW

File: ui/shared/BugFiler.jsx [hunk 1]
```diff
@@ -536,6 +543,35 @@ export class BugFilerClass extends React.Component {
     });
   };
 
+  checkForSecurityIssue() {
+    if (this.state.isSecurity) {
+      return;
+    }
+
+    const inputToCheck = `${this.state.summary}\n${this.state.comment}`;
+
```
Comment (sarah-clements) [inline ui/shared/BugFiler.jsx:552]: You could unpack the state values: `const { summary, comment, isSecurity } = this.state;`

File: ui/shared/BugFiler.jsx [hunk 2]
```diff
@@ -167,6 +167,7 @@ export class BugFilerClass extends React.Component {
       isFilerSummaryVisible: false,
       selectedProduct: null,
       isIntermittent: true,
+      isSecurity: false,
```
Comment (sarah-clements) [inline ui/shared/BugFiler.jsx:170]: I think it'd be clearer if this was `isSecurityIssue`.

File: ui/shared/BugFiler.jsx [hunk 3]
```diff
@@ -177,6 +178,10 @@ export class BugFilerClass extends React.Component {
     };
   }
 
+  componentDidMount() {
+    this.checkForSecurityIssue();
```
Comment (sarah-clements) [inline ui/shared/BugFiler.jsx:182]: Why is this called when the component mounts?

**Review by sarah-clements — COMMENTED** [review]
Looks good, just a few nits and question :)

File: ui/shared/BugFiler.jsx (hunk 3 above)
Comment (Archaeopteryx) [reply ui/shared/BugFiler.jsx:182]: When the bugfiler gets rendered with an initial summary which indicates a security issue, the summary needs to be evaluated for this (comment is always blank on render).

**Review by sarah-clements — APPROVED** [review]
Looks good, thanks!

**Pull request merged**